browser/caliban.py

```diff
@@ -363,237 +400,195 @@ def action_threshold(self, y1, x1, y2, x2, frame, label):
                        left_edge:right_edge, self.feature] = safe_overlay
 
         # don't need to update cell_info unless an annotation has been added
-        if np.any(np.isin(self.annotated[frame, :, :, self.feature], label)):
-            self.add_cell_info(feature=self.feature, add_label=label, frame=frame)
+        if np.any(np.isin(self.annotated[frame, ..., self.feature], label)):
+            self.add_cell_info(add_label=label, frame=frame)
 
-    def action_flood_contiguous(self, label, frame, x_location, y_location):
-        '''
-        flood fill a cell with a unique new label; alternative to watershed
-        for fixing duplicate label issue if cells are not touching
-        '''
-        img_ann = self.annotated[frame, :, :, self.feature]
-        old_label = label
-        new_label = np.max(self.cell_ids[self.feature]) + 1
 
-        in_original = np.any(np.isin(img_ann, old_label))
+class ZStackReview(BaseReview):
 
-        filled_img_ann = flood_fill(img_ann, (y_location, x_location), new_label)
-        self.annotated[frame, :, :, self.feature] = filled_img_ann
+    def __init__(self, filename, input_bucket, output_bucket, subfolders, rgb=False):
+        super(ZStackReview, self).__init__(
+            filename, input_bucket, output_bucket, subfolders,
+            raw_key='raw', annotated_key='annotated')
 
-        in_modified = np.any(np.isin(filled_img_ann, old_label))
+        self.rgb = rgb
 
-        # update cell info dicts since labels are changing
-        self.add_cell_info(feature=self.feature, add_label=new_label, frame=frame)
+        if self.rgb:
+            # possible differences between single channel and rgb displays
+            if self.raw.ndim == 3:
```

Comment:
this happens after `self.width` and `self.height` are assigned and could cause problems. Maybe reassign the size attributes if the dimensions are updated? Otherwise I would expect a (512, 512, 3) image to be have "width" of 2 and 512 "frames".

Reply:
Would just assigning the width and height again resolve the problem?

```python
    if self.raw.ndim == 3:
            self.raw = np.expand_dims(self.raw, axis=0)
            self.annotated = np.expand_dims(self.annotated, axis=0)

            self.rgb_img = self.reduce_to_RGB()

            self.max_frames = self.raw.shape[0]
            self.height = self.raw.shape[1]
            self.width = self.raw.shape[2]
```

Reply:
Yes, I think that would fix the issue.